sync/src/main/java/org/bf2/sync/ManagedKafkaSync.java

```diff
@@ -231,7 +245,12 @@ void reconcile(String remoteId, String localMetaNamespaceKey) {
                     ObjectMeta meta = remoteCopy.getMetadata();
                     client.edit(local.getMetadata().getNamespace(), local.getMetadata().getName(), mk -> {
                             mk.getMetadata().setLabels(meta.getLabels());
+                            Map<String, String> existingAnnotations = mk.getMetadata().getAnnotations();
                             mk.getMetadata().setAnnotations(meta.getAnnotations());
+                            if (existingAnnotations != null) {
+                                existingAnnotations.keySet().retainAll(DATA_PLANE_ANNOTATIONS);
+                                mk.getMetadata().getAnnotations().putAll(existingAnnotations);
```

Comment:
I am failing to read this (never used retainAll method before), so we are removing all the annotations from existing MK except for DATA_PLANE_ANNOTATIONS, and updating the resource. What happens to any other annotations that are set by CP or Kube?